src/qz/ui/component/CertificateTable.java

```diff
@@ -0,0 +1,155 @@
+        TRUSTED("Trusted", "isTrusted"),
+        VALID_FROM("Valid From", "getValidFrom"),
+        VALID_TO("Valid To", "getValidTo"),
+        FINGERPRINT("Fingerprint", "getFingerprint");
+
+        String description;
+        String callBack;
+
+        CertificateField(String description, String callBack) {
+            this.description = description;
+            this.callBack = callBack;
+        }
+
+        /**
+         * Returns the <code>String</code> value associated with this certificate field
+         *
+         * @return Certificate field such as "commonName"
+         */
+        public String getValue(Certificate cert) {
+            if (cert == null) {
+                return "";
+            }
+
+            Reflect reflect = Reflect.on(cert).call(callBack);
+            Object value = reflect == null? null:reflect.get();
+            if (value == null) {
+                return "";
+            }
+            return value.toString();
+        }
+
+        @Override
+        public String toString() {
+            return description;
+        }
+
+        public String getDescription() {
+            return description;
+        }
+
+        public static int size() {
+            return values().length;
+        }
+    }
+
+    private Certificate cert;
+
+    private Instant warn;
+    private Instant now;
+
+    public CertificateTable(IconCache iconCache) {
+        super(iconCache);
+        setDefaultRenderer(Object.class, new CertificateTableCellRenderer());
+    }
+
+    public void setCertificate(Certificate cert) {
+        this.cert = cert;
+        refreshComponents();
+    }
+
+    @Override
+    public void refreshComponents() {
+        if (cert == null) {
+            return;
+        }
+
+        now = Instant.now();
+        warn = now.plus(Constants.EXPIRY_WARN, ChronoUnit.DAYS);
+
+        removeRows();
+
+        // First Column
+        for(CertificateField field : CertificateField.values()) {
+            model.addRow(new Object[] {field, field.getValue(cert)});
+        }
+
+        repaint();
+    }
+
+    @Override
+    public void refresh() {
+        refreshComponents();
+        ((StyledTableCellRenderer)getDefaultRenderer(Object.class)).refresh();
+    }
+
+    public void autoSize() {
+        super.autoSize(CertificateField.size(), 2);
+    }
+
+
+    /** Custom cell renderer for JTable to allow colors and styles not directly available in a JTable */
+    private class CertificateTableCellRenderer extends StyledTableCellRenderer {
+
+        @Override
+        public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int col) {
+            JLabel label = (JLabel)super.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, col);
+
+            // First Column
+            if (value instanceof CertificateField) {
+                stylizeLabel(STATUS_NORMAL, label, isSelected);
```

Comment:
You don't explicitly return the `label` here.  Was this intended?  It's not consistent.

Reply:
`stylizeLabel` just returns the label object passed in the parameters, but we are setting an icon in this block first and then returning it.

Reply:
Sorry, I'm talking about the lack of redundant assignment (`return` was in context to the function being called )